### ISSUE TITLE
`autoWebview` loop until timeout

### ISSUE THREAD

**Issue by lionelgaillard** [appium/appium#10784]
## The problem

When testing an Ionic app on an Android device, the `autoWebview` option fails to set correct context. It seems able to found the webview and available contexts, but continue to ask for "a list of available webviews" indefinitely until the timeout exceeds :

```
[debug] [AndroidDriver] Getting a list of available webviews
[debug] [ADB] Running '/Users/<user>/Library/Android/sdk/platform-tools/adb -P 5037 -s 11160ba3419f3303 shell cat /proc/net/unix'
[debug] [AndroidDriver] Found webviews: ["WEBVIEW_Terrace"]
[debug] [AndroidDriver] Available contexts: ["NATIVE_APP","WEBVIEW_Terrace"]
[debug] [AndroidDriver] Getting a list of available webviews
[debug] [ADB] Running '/Users/<user>/Library/Android/sdk/platform-tools/adb -P 5037 -s 11160ba3419f3303 shell cat /proc/net/unix'
[debug] [AndroidDriver] Found webviews: ["WEBVIEW_Terrace"]
[debug] [AndroidDriver] Available contexts: ["NATIVE_APP","WEBVIEW_Terrace"]
[debug] [AndroidDriver] Getting a list of available webviews
[debug] [ADB] Running '/Users/<user>/Library/Android/sdk/platform-tools/adb -P 5037 -s 11160ba3419f3303 shell cat /proc/net/unix'
[debug] [AndroidDriver] Found webviews: ["WEBVIEW_Terrace"]
[debug] [AndroidDriver] Available contexts: ["NATIVE_APP","WEBVIEW_Terrace"]
[debug] [AndroidDriver] Getting a list of available webviews

// again and again until timeout (link to full log below)...
```

> Note: I have no idea why the webview is called `WEBVIEW_Terrace`

Then, the timeout exceeds, it closes the session and ends to a "No such context found" error :

```
[16:59:11] E/launcher - Error: WebDriverError: No such context found.
    at Object.checkLegacyResponse (/Users/<user>/<path>/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/Users/<user>/<path>/node_modules/selenium-webdriver/lib/http.js:508:13)
    at doSend.then.response (/Users/<user>/<path>/node_modules/selenium-webdriver/lib/http.js:440:30)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:118:7)
```

## Environment

* Appium version (or git revision) that exhibits the issue: 1.8.0
* Last Appium version that did not exhibit the issue (if applicable): -
* Desktop OS/version used to run Appium: macOS 10.13.3
* Node.js version (unless using Appium.app|exe): 9.9.0
* Mobile platform/version under test: Android 7
* Real device or emulator/simulator: Real device
* Appium CLI or Appium.app|exe: Appium CLI

## Details

I already encounter this issue in the past, but at this time, I've been able to workaround by setting `autoWebview` to `false` and set the context manually before my tests: 

    driver.contexts(contexts => {
        driver.context(contexts[1]);
    });

> (the context's name was changing, but was always the second one)

Today I have to use Protractor and I can no longer turn off `autoWebview` or I'll get a `Not implemented yet for script` error:
```
[16:12:45] E/launcher - Error: WebDriverError: Not implemented yet for script.
    at Object.checkLegacyResponse (/Users/buildserver/Inextrace/Mobile/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/Users/buildserver/Inextrace/Mobile/node_modules/selenium-webdriver/lib/http.js:508:13)
    at doSend.then.response (/Users/buildserver/Inextrace/Mobile/node_modules/selenium-webdriver/lib/http.js:440:30)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:118:7)
```
I found it was caused because Protractor was trying to set its `allScriptsTimeout` (default to 11000) against the `NATIVE_APP` context, and this option value can be changed but not disabled.

So basically `autoWebview` can't be turned off with Protractor.

## Link to Appium logs

[Full log](https://gist.github.com/lionelgaillard/2ee43716e19b6acacf78050873160f2b)

## Code To Reproduce Issue [ Good To Have ]

Capabilities:

    capabilities: {
        browserName: '',
        app: __dirname + '/../build/dist/demo/android-release.apk',
        bundleId: 'com.xxx',
        deviceName: 'android',
        platformName: 'Android',
        platformVersion: '7',
        automationName: 'UiAutomator2',
        appPackage: 'com.xxx',
        appActivity: '.MainActivity',
        autoWebview : true,
        autoWebviewTimeout: 20000,
        skipUnlock: true,
    }

**Comment by aj-dev** [comment]
I'm having the same issue with Cordova app and `autoWebview: true` on Android emulator. It finds 2 available contexts but never selects the webview

**Comment by imurchie** [comment]
The webview, and its name, is retrieved from the process list. Could you provide the output from running `adb shell cat /proc/net/unix`?

**Comment by imurchie** [comment]
I've found the problem. A fix is in the works.

**Comment by imurchie** [comment]
The latest beta of Appium ought to work better.

**Comment by mayureshshirodkar** [comment]
@imurchie is this fix available in the latest Appium beta, since running on android creates this: 
`[Appium] Creating new AndroidUiautomator2Driver (v1.10.0) session`
instead of using` appium-uiautomator2-driver@1.13.2`.

Log:https://gist.github.com/mayureshshirodkar/b6ad7ab91de65796fa4a91a4b6a9026e
